DOCSP-42743 Collection config (#95)
mongodb/docs-mongoid: source/configuration/collection-config.txt

.. _mongoid-collection-config:

========================
Collection Configuration
========================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: code example, collections, time series, capped collection

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------

In this guide, you can learn how to specify configuration options for a
collection in your {+odm+} application.

Configure Collection Options
----------------------------

You can specify configuration options for a collection by using the
``:collection_options`` argument with the ``store_in``
macro. The ``:collection_options`` argument accepts any collection option that

> your driver and server version supports. 

Reviewer: ```suggestion
your {+ruby-driver+} and MongoDB server version supports. 
```

.. note::

>    To apply the collection options, your collection must be explicitly created
>    prior to use. To learn how to create a collection using a Rake task, see the

Reviewer: ```suggestion
   You must explicitly create a collection before you apply any collection options. To learn
   how to create a collection by using a Rake task, see the
```

>    :ref:`mongoid-create-collection-rake` section of this guide.

Reviewer: S: consider linking to docs in server or atlas about creating collections.

Author: Ended up rewording this section to better match what the original docs were saying. It says that it "should be done" by the Rake task so it seems to me like they want to point users that way rather than other ways of creating collection

To learn more about collection options available in the {+ruby-driver+}, see the
:ruby:`Collections </reference/collection-tasks/>` guide in the {+ruby-driver+}
documentation.

> The following sections show examples of how to configure collection options with
> {+odm+}.

Reviewer: ```suggestion
The following sections show examples of how to configure collection options when using
{+odm+}.
```

Time Series Collection
~~~~~~~~~~~~~~~~~~~~~~

Time series collections efficiently store sequences of measurements over a
period of time. The following example shows how to configure a time series

> collection with
> {+odm+}:

Reviewer: ```suggestion
collection:
```

> {+odm+}:

Reviewer: S: could be helpful to describe what the options in the code are doing. Or you can link out to docs on time series collections or both ideally

Author: I linked out to the manual for descriptions since the functionality of the collection isn't the focus of this page

.. literalinclude:: /includes/configuration/collection-config.rb
   :language: ruby
   :start-after: # start-time-series-config

>    :end-before: # end-time-series-config

Reviewer: S: consider emphasizing lines

Capped Collection
~~~~~~~~~~~~~~~~~

> Capped collections have maximum size or document counts that prevent them from
> growing beyond maximum thresholds. The following example shows how to configure
> a capped collection with {+odm+}:

Reviewer: ```suggestion
Capped collections have maximum size or document counts that prevent them from
growing beyond a specified threshold. The following example shows how to configure
a capped collection:
```

> 

Reviewer: S: same as in the prev section, consider linking to docs about capped collections + explaining what the options do

.. literalinclude:: /includes/configuration/collection-config.rb
   :language: ruby
   :start-after: # start-capped-collection-config
   :end-before: # end-capped-collection-config

Default Collation
~~~~~~~~~~~~~~~~~

Collations are sets of rules for how to compare strings, typically in a
particular natural language. The following example shows how to specify a
default collation to use on a
collection with {+odm+}:

> 

Reviewer: S: same comment about explaining + linking
